### PR TITLE
Submit nearest form even if not direct descendent

### PR DESCRIPTION
**Pull request by troygnichols**
Hi! I'm not positive this is the right thing to do, but it solved a problem for me.

Say I have a template like this:

```
<%= form_for ... do %>
  <div id="foo">
    <%= link to: "/somewhere", class: "fa fa-save", data: [submit: "parent"] %>
  </div>
<% end %>
```

(I use `<a>` instead of `<submit>`  because I need to style the submit button with font-awesome icons e.g.)

With the original code, it tried to submit its immediate parent`div#foo`, which of course doesn't work. This PR looks for the nearest parent that is a form and submits that.

### CODE REVIEW

**Comment by josevalim** [comment]
@troygnichols unfortunately closest is not supported on internet explorer and on mobiles. :(

**Comment by troygnichols** [comment]
@josevalim Ah, okay. I've updated to use no-frills JS now.

File: priv/static/phoenix_html.js [hunk 1]
```diff
@@ -7,12 +7,22 @@ function isLinkToSubmitParent(element) {
   return isLinkTag && shouldSubmitParent;
 }
 
+function getClosestForm(element) {
+  while (element && element !== document && element.nodeType === Node.ELEMENT_NODE) {
+    if (element.tagName.toLowerCase() === 'form') {
```
Comment (josevalim) [inline priv/static/phoenix_html.js]: We should use `element.tagName === 'FORM'` to follow the same pattern we used in the function above.

**Comment by troygnichols** [comment]
That makes sense. Updated.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**